src/query-builder/SelectQueryBuilder.ts

```diff
@@ -2034,6 +2034,26 @@ export class SelectQueryBuilder<Entity extends ObjectLiteral>
         joinAttribute.condition = condition // joinInverseSideCondition
         // joinAttribute.junctionAlias = joinAttribute.relation.isOwning ? parentAlias + "_" + destinationTableAlias : destinationTableAlias + "_" + parentAlias;
         this.expressionMap.joinAttributes.push(joinAttribute)
+        const isEntity = this.connection.hasMetadata(entityOrProperty)
+        const isSubQuery =
+            (!isEntity && typeof entityOrProperty === "function") ||
```

Comment:
Class references will give `function` when calling `typeof foo`. Now the argument is determined to be an entity if the connection has metadata.